### DISCUSSION TITLE
Slash or no slash?

### DISCUSSION THREAD

**Discussion by rmwiseman** [bluenviron/mediamtx#620] [Questions]
I've set up an on-demand `ffmpeg` command, but when I try to access it, the RTSP Simple Server log shows:
```
2021/10/04 10:51:29 I [0/0] [RTSP] [conn 10.30.210.1:42690] ERR: path must end with a slash (robot)
```
So I then try to access it with a slash at the end of the path and the log shows:
```
2021/10/04 10:51:48 I [0/0] [RTSP] [conn 10.30.210.1:42700] ERR: invalid path name: can't end with a slash (robot/)
```
So which is it?  I'm very confused!

**EDIT:** This was when I was using the docker version of RTSP Simple Server.  I think there's an underlying issue where that doesn't actually set up on-demand `ffmpeg` execution.  I'm not even sure that `ffmpeg` is installed on the docker image.  I tried to log into a shell, and /bin/sh isn't even installed.  When I run in the "standard" way (i.e. not using docker) it works fine.

**Comment by aler9** [comment]
RTSP SETUP requests must contains a path that ends with a slash, but it's the player responsibility to add it. A user must always use paths without a final slash. What happens is the following:

1. VLC connects to the server
2. the server tries to start ffmpeg, it fails since you're using a docker image that doesn't contain ffmpeg
3. VLC times out, switches to an old version of RTSP and sends a SETUP request without a slash

There are plenty of discussions about the issue here, it's a VLC behavior.

**Reply by rmwiseman** [reply]
Thanks, but as my edit suggests, I think I got this error because nothing was publishing on the path.  I think it might show this error in error!

[a comment by an automated account is not shown]
